extras/indexing/src/main/java/org/apache/rya/indexing/entity/query/EntityQueryNode.java

```diff
@@ -0,0 +1,356 @@
+            }
+        }
+    }
+
+    /**
+     * Verifies a single Statement Pattern defines the Type of Entity this query node matches.
+     *
+     * @param type - The expected Type. (not null)
+     * @param patterns - The patterns to check. (not null)
+     * @throws IllegalStateException No Type or the wrong Type is specified by the patterns.
+     */
+    private static void verifyHasCorrectTypePattern(final Type type, final Collection<StatementPattern> patterns) throws IllegalStateException {
+        requireNonNull(type);
+        requireNonNull(patterns);
+
+        boolean typeFound = false;
+
+        for(final StatementPattern pattern : patterns) {
+            final RyaURI predicate = new RyaURI(pattern.getPredicateVar().getValue().toString());
+
+            if(predicate.equals(TYPE_ID_URI)) {
+                final RyaURI typeId = new RyaURI( pattern.getObjectVar().getValue().stringValue() );
+                if(typeId.equals(type.getId())) {
+                    typeFound = true;
+                } else {
+                    throw new IllegalStateException("Statement Pattern encountred for a Type that does not match the expected Type." +
+                            " Expected Type = '" + type.getId().getData() + "' Found Type = '" + typeId.getData() + "'");
+                }
+            }
+        }
+
+        if(!typeFound) {
+            throw new IllegalStateException("The collection of Statement Patterns that this node matches must define which Type they match.");
+        }
+    }
+
+    /**
+     * Verify all of the patterns have predicates that match one of the Type's property names.
+     *
+     * @param type - The Type the patterns match. (not null)
+     * @param patterns - The patterns to check.
+     * @throws IllegalStateException If any of the non-type defining Statement Patterns
+     *   contain a predicate that does not match one of the Type's property names.
+     */
+    private static void verifyAllPredicatesPartOfType(final Type type, final Collection<StatementPattern> patterns) throws IllegalStateException {
+        requireNonNull(type);
+        requireNonNull(patterns);
+
+        for(final StatementPattern pattern : patterns) {
+            // Skip TYPE patterns.
+            final RyaURI predicate = new RyaURI( pattern.getPredicateVar().getValue().toString() );
+            if(predicate.equals(TYPE_ID_URI)) {
+                continue;
+            }
+
+            if(!type.getPropertyNames().contains(predicate)) {
+                throw new IllegalStateException("The Predicate of a Statement Pattern must be a property name for the Type. " +
+                        "Type ID: '" + type.getId().getData() + "' Pattern: " + pattern);
+            }
+        }
+    }
+
+    @Override
+    public CloseableIteration<BindingSet, QueryEvaluationException> evaluate(final Collection<BindingSet> bindingSets) throws QueryEvaluationException {
+        // TODO Auto-generated method stub
+        return null;
```

Comment:
I didn't see it getting called from anywhere, its an artifact of what kchilton2 did.  No idea what would go in here